spec/Section 2 -- Source Schema.md

```diff
@@ -2,59 +2,120 @@
 
 ## Directives
 
-### @entityResolver
+### @lookup
 
 ```graphql
-directive @entityResolver on FIELD_DEFINITION
+directive @lookup on FIELD_DEFINITION
 ```
 
-Entity resolvers are fields on the query root type of a subgraph that can
-resolve an entity by a stable key. The stable key is defined by the arguments of
-the field.
+The `@lookup` directive is used within a _source schema_ to specify object
+fields that can be used by the _distributed GraphQL executor_ to resolve an
+entity by a stable key.
+
+The stable key is defined by the arguments of the field. Only fields that are
+annotated with the `@lookup` directive will be recognized as lookup field.
+
+Source schemas can provide multiple lookup fields for the same entity with
+different keys.
+
+In this example the source schema specifies that the `Person` entity can be
+resolved with the `personById` field or the `personByName` field on the `Query`
+type. Both fields can resolve the same entity but do so with different keys.
 
 ```graphql example
 extend type Query {
-  version: Int # NOT an entity resolver.
-  personById(id: ID!): Person @entityResolver
+  version: Int # NOT a lookup field.
+  personById(id: ID!): Person @lookup
```

Comment:
should we also consider the upcoming `@oneOf` directive support as well? i.e. this could lead to some potentially interesting cases like

```graphql
input PersonBy @oneOf {
  id: ID
  name: String
}
type Query {
  person(by: PersonBy!): Person @lookup
}
```

Reply:
This is covered ... but worked on in a separate RFC with the is directive.